pymc/model.py

```diff
@@ -2035,10 +2035,43 @@ def Deterministic(name, var, model=None, dims=None):
 def Potential(name, var, model=None):
     """Add an arbitrary factor potential to the model likelihood
 
+      Potential function is used to add arbitrary factors i.e. constraints that we define to the likelihood function of a model to adjust the probablity density of the model. 
+
+    .. code:: python
+
+        with pm.Model() as my_model:
+          x = pm.Normal('x', mu=0, tau=1)
+          y = pm.Normal('y', mu=x, tau=1)
+
+          constraint = x + y < 1
+          potential = pm.Potential('constraint', constraint, my_model)
+
+        In this example, we define a constraint on the sum of x and y that it can only be lesser than 1. We pass this constraint to the pm.Potential function.
+        Now the pm.Potential function takes in a expression, a scalar value and the model name as argument.
+        Now, the probablity density that the model produces takes into consideration the constraint that the sum of x and y should be lesser than 1.
+
+        The Potential function here acts as a gatekeeper allowing only samples that satisfy the constraint to contribute to the likelihood. This ensures that the sample generated by the sampler are consistent with the constraint.
+
+        Have a look at the following example:
+       
+     .. code:: python
+
+        with pm.Model() as model:
+          x = pm.Normal('x', mu=0, sigma=1)
+          y = pm.Normal('y',mu=x, sigma=1,observed=data)
+          constraint = x>= 0
+          potential = pm.Potential('x_constraint',-tt.log(tt.switch(constraint, 1, 1e-30)))
+
+        In this example, we define a constraint on x that it can be either greater than or equal to 0 and pass this constraint to the pm.Potential function.
+        Now the pm.Potential function takes in a string expression and a scalar value as argument. Hence we pass '-tt.log(tt.switch(constraint, 1, 1e-30))' as second argument which will return a scalar value depending on the constraint and will add to the likelihood function of the model.
+        This is done because PyMC3 uses an energy-based model, which means that the probability of a variable configuration is proportional to the negative of the energy of that configuration. So, the negative logarithm of the switch function ensures that the energy is high (i.e. close to negative infinity) when the constraint is not met, and low (i.e. close to zero) when the constraint is met.
```

Comment:
```suggestion
The negative logarithm of the switch function ensures that this component of the log-likelihood is high (i.e. close to 30) when the constraint is not met, and low (i.e. close to zero) when the constraint is met.
```

I'm confused with this sentence. What do you mean by "probability of a variable configuration"?

Lastly, personally, I would avoid using the energy concept since it's similar to the likelihood and focus on the log-likelihood since that's what PyMC works with. In the sense that I would avoid introducing too many new concepts as once, although the energy-based analogy is correct (to my understanding and despite my little knowledge of physics 😅). 

However, with that, the sentence would revert the direction of the logic if I'm not mistaken...

Reply:
Answering the first question,  'probability of variable configuration' refers to the likelihood function. By this I simply  mean to refer the likelihood of a particular set of values for the variables that the model equation uses.

For the second part, no it does not revert the direction of logic... That's bad framing on my part😅

What I meant from "close to  negative infinity" was that the log function/ log-likelihood will be close to negative infinity. The mathematical negative will thus be a large positive value. Energy is the negative logarithmic function that we have defined and thus it will be high when the constraint is not met. 

Can we re-frame this as:
The reciprocal exponential function of the negative logarithm is proportional to the likelihood function. When the constraint is met, the switch function returns 1 making the negative logarithm 0 and hence likelihood function has no penalizing effect. When the constraint is not met the switch function returns 1e-30 and the negative logarithm takes a high value(i.e. 30), reciprocal exponential function of which makes the likelihood function very small and successfully penalizes it.

Reply:
Sorry, chipping after much time. I'm not sure what you mean by "reciprocal exponential function of the negative logarithm"... 

However, it looks like you have it figured out in some threads below! Perhaps you can focus on those suggestions rather than on my comments